internal/pkg/agent/application/upgrade/details/state.go

```diff
@@ -0,0 +1,20 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package details
+
+type State string
+
+const (
+	StateRequested   State = "UPG_REQUESTED"
```

Comment:
Yes, there's a bit of foreshadowing going on here: as per the design of the upgrade tracking project, these state names,  `UPG_REQUESTED`, `UPG_DOWNLOADING`, etc. are the exact names being expected by Fleet Server and the UI.  The UI will likely make decisions based on these names, e.g. to show the download progress percentage when the upgrade state is `UPG_DOWNLOADING` and will also likely have translations into more user-friendly text based on these names.  So rather than have an enumeration here, only to then have a small translation/lookup from enumerated value to the string values somewhere, I decided to just make the state values directly be the string values we need.

The downsides you bring up are valid but I'm not sure addressing them is worth sacrificing the practical aspect I mentioned above.  Happy to be convinced otherwise, though — I don't feel too strongly about this one!

Reply:
It would be nice if we could generate these from the Fleet OpenAPI definition of them: https://github.com/elastic/fleet-server/blob/140180c6bbaebe88c8463beeb717cacb036a68e3/model/openapi.yml#L294-L321

I'm not sure how much additional work this is, if its lots it isn't worth blocking this implementation on. We can do that as a follow up since we probably want to consume more than just these states.

Reply:
We could at least link back to that API specification from here so it's easy to manually check whether the two are in sync.

Reply:
That's a great idea.  I added a task in https://github.com/elastic/elastic-agent/issues/3119#issuecomment-1743638526 to implement this suggestion.

Reply:
> We could at least link back to that API specification from here so it's easy to manually check whether the two are in sync.

Done in 329e599bf13b87a362d5954081888f0b69789338.